src/config.rs

```diff
@@ -356,7 +356,12 @@ pub fn construct_config(pc: ProxyConfig) -> Result<Config, Error> {
     // Increase some defaults. Note these are NOT coming from /etc/resolv.conf (only some fields do, we don't override those),
     // but rather hickory's hardcoded defaults
     dns_resolver_opts.cache_size = 4096;
-    // TODO: should we override server_ordering_strategy based on our IP support?
+    dns_resolver_opts.ip_strategy = if ipv6_enabled {
+        // Lookup both in parallel. We will do filtering in a later stage to only appropriate IP families
+        LookupIpStrategy::Ipv4AndIpv6
```

Comment:
Maybe a note that this is needed rather than `Ipv6thenIpv4` so that we can match the ip family of the original request when ipv6 is enabled rather than blanket using ipv6 whenever AAAA records exist. I did spend a sec wondering if we could optimize this using a better strategy.

Reply:
I expanded the comment